UPGRADE.md

```diff
@@ -0,0 +1,674 @@
+# Upgrading from Curator v3 to v4
+
+This guide will walk you through upgrading from Filament Curator v3 to v4. This is a major version upgrade that includes significant breaking changes due to the upgrade from Filament v3 to v4 and Tailwind CSS v3 to v4.
+
+## Requirements
+
+Before upgrading, ensure your project meets these requirements:
+
+- PHP 8.2 or higher
+- Laravel 11.28 or higher (required by Filament 4)
+- Filament 4.0 or higher
+- Tailwind CSS 4.0 or higher
+
+## Overview of Major Changes
+
+Curator v4 brings significant architectural improvements and breaking changes:
+
+### Framework Upgrades
+- **Filament**: v3.x → v4.0+
+- **Tailwind CSS**: v3.x → v4.x
+- **Asset Management**: Complete overhaul using Tailwind 4's new system
+
+### Configuration Restructuring
+- Flattened config structure reorganized into nested groups
+- Many config keys renamed or relocated
+- New required environment variables
+
+### Installation Simplification
+- Removed cropper.js dependency (cropper.css no longer needed)
+- Curator's component CSS still required via `@import`
+- Streamlined asset registration (CSS no longer registered in ServiceProvider)
+
+### Code Architecture
+- New manager classes for better organization
+- Facade pattern introduced
+- Enums for type safety
+- DTOs for data transfer
+
+## Step-by-Step Upgrade Guide
+
+### Step 1: Update Composer Dependencies
+
+Update your `composer.json`:
+
+```bash
+composer require awcodes/filament-curator:^4.0 filament/filament:^4.0
+```
+
+### Step 2: Remove cropper.js Dependency
+
+> **Important Note**: Unlike the v2→v3 upgrade, there is **no database migration** required for v3→v4. The database schema remains the same. The `curator:upgrade` command (which was for v2→v3) does not exist in v4.
```

Comment:
Not sure why this is under the cropper.js section, but there is a change to the database table name in v4 to 'curator'. There should probably be a dedication section about either changing the name of the table in the database or overriding the Media model to change the `$table` property to whatever it is in their current app.

Reply:
Kindly check the updated Step 2: Database migration and review.